lib/ex_aws/request/url.ex

```diff
@@ -17,6 +18,16 @@ defmodule ExAws.Request.Url do
     |> String.trim_trailing("?")
   end
 
+  defp set_host(config, operation) do
+    case operation do
+      %ExAws.Operation.S3{} ->
+        %{config | host: "#{operation.bucket}.s3.amazonaws.com"}
+
+      _ ->
+        %{config | host: config.host}
+    end
+  end
```

Comment:
For a function that's just one `case` operation, I think it's preferable to divide it into two pattern matches like this:

```elixir
defp set_host(config, %ExAws.Operation.S3{bucket: bucket}), do: %{config | host: "#{bucket}.s3.amazonaws.com"}

defp set_host(config, _operation), do: config
```